central/reports/scheduler/v2/reportgenerator/report_gen_impl.go

```diff
@@ -148,8 +142,16 @@ func (rg *reportGeneratorImpl) generateReportAndNotify(req *ReportRequest) error
 				errorList.AddError(errors.Errorf("incorrect type of notifier '%s'", notifierSnap.GetEmailConfig().GetNotifierId()))
 				continue
 			}
-			err := rg.retryableSendReportResults(reportNotifier, notifierSnap.GetEmailConfig().GetMailingLists(),
-				zippedCSVData, messageText)
+			emailBodyStrNotifier := notifierSnap.GetEmailConfig().GetCustomBody()
+			if notifierSnap.GetEmailConfig().GetCustomBody() == "" {
+				emailBodyStrNotifier, err = formatMessage(req.DataStartTime, templateStr, req.ReportSnapshot.GetVulnReportFilters().GetImageTypes())
+				if err != nil {
+					return errors.Wrap(err, "error formatting the report email text")
+				}
+			}
+
+			err = rg.retryableSendReportResults(reportNotifier, notifierSnap.GetEmailConfig().GetMailingLists(),
+				zippedCSVData, emailBodyStrNotifier, notifierSnap.GetEmailConfig().GetCustomSubject())
```

Comment:
If default body is stored in messageText outside of the for loop,

```suggestion
			customBody := notifierSnap.GetEmailConfig().GetCustomBody()
			if customBody != nil {
			    messageText = customBody
			}
			
			err = rg.retryableSendReportResults(reportNotifier, notifierSnap.GetEmailConfig().GetMailingLists(),
				zippedCSVData, customBody, notifierSnap.GetEmailConfig().GetCustomSubject())
```